config/config_test.go

```diff
@@ -124,6 +125,154 @@ func TestDecodeConfig(t *testing.T) {
 		"Did not load pipeline config correctly")
 }
 
+func TestDecodeConfigWithEnv(t *testing.T) {
+	const extensionExtra = "some extension string"
+	const extensionExtraMapValue = "some extension map value"
+	const extensionExtraListElement = "some extension list element"
+	os.Setenv("EXTENSIONS_EXAMPLEEXTENSION_EXTRA", extensionExtra)
+	os.Setenv("EXTENSIONS_EXAMPLEEXTENSION_EXTRA_MAP_EXT_VALUE", extensionExtraMapValue)
+	os.Setenv("EXTENSIONS_EXAMPLEEXTENSION_EXTRA_LIST_ELEMENT_1", extensionExtraListElement)
+
+	const receiverExtra = "some receiver string"
+	const receiverExtraMapValue = "some receiver map value"
+	const receiverExtraListElement = "some receiver list element"
+	os.Setenv("RECEIVERS_EXAMPLERECEIVER_EXTRA", receiverExtra)
+	os.Setenv("RECEIVERS_EXAMPLERECEIVER_EXTRA_MAP_RECV_VALUE", receiverExtraMapValue)
+	os.Setenv("RECEIVERS_EXAMPLERECEIVER_EXTRA_LIST_ELEMENT_1", receiverExtraListElement)
+
+	const processorExtra = "some processor string"
+	const processorExtraMapValue = "some processor map value"
+	const processorExtraListElement = "some processor list element"
+	os.Setenv("PROCESSORS_EXAMPLEPROCESSOR_EXTRA", processorExtra)
+	os.Setenv("PROCESSORS_EXAMPLEPROCESSOR_EXTRA_MAP_PROC_VALUE", processorExtraMapValue)
+	os.Setenv("PROCESSORS_EXAMPLEPROCESSOR_EXTRA_LIST_ELEMENT_1", processorExtraListElement)
+
+	const exporterExtra = "some exporter string"
+	const exporterExtraMapValue = "some exporter map value"
+	const exporterExtraListElement = "some exporter list element"
+	os.Setenv("EXPORTERS_EXAMPLEEXPORTER_EXTRA_INT", "65")
+	os.Setenv("EXPORTERS_EXAMPLEEXPORTER_EXTRA", exporterExtra)
+	os.Setenv("EXPORTERS_EXAMPLEEXPORTER_EXTRA_MAP_EXP_VALUE", exporterExtraMapValue)
+	os.Setenv("EXPORTERS_EXAMPLEEXPORTER_EXTRA_LIST_ELEMENT_1", exporterExtraListElement)
+
+	defer func() {
+		os.Unsetenv("EXTENSIONS_EXAMPLEEXTENSION_EXTRA")
+		os.Unsetenv("EXTENSIONS_EXAMPLEEXTENSION_EXTRA_MAP_EXT_VALUE")
+		os.Unsetenv("EXTENSIONS_EXAMPLEEXTENSION_EXTRA_LIST_ELEMENT_1")
+
+		os.Unsetenv("RECEIVERS_EXAMPLERECEIVER_EXTRA")
+		os.Unsetenv("RECEIVERS_EXAMPLERECEIVER_EXTRA_MAP_RECV_VALUE")
+		os.Unsetenv("RECEIVERS_EXAMPLERECEIVER_EXTRA_LIST_ELEMENT_1")
+
+		os.Unsetenv("PROCESSORS_EXAMPLEPROCESSOR_EXTRA")
+		os.Unsetenv("PROCESSORS_EXAMPLEPROCESSOR_EXTRA_MAP_PROC_VALUE")
+		os.Unsetenv("PROCESSORS_EXAMPLEPROCESSOR_EXTRA_LIST_ELEMENT_1")
+
+		os.Unsetenv("EXPORTERS_EXAMPLEEXPORTER_EXTRA_INT")
+		os.Unsetenv("EXPORTERS_EXAMPLEEXPORTER_EXTRA")
+		os.Unsetenv("EXPORTERS_EXAMPLEEXPORTER_EXTRA_MAP_EXP_VALUE")
+		os.Unsetenv("EXPORTERS_EXAMPLEEXPORTER_EXTRA_LIST_ELEMENT_1")
+	}()
+
+	t.Log(expandStringValues([]string{"${EXTENSIONS_EXAMPLEEXTENSION_EXTRA}"}))
```

Comment:
Is this still necessary?

Reply:
as I said, I want to see how people feel about this, I need more work to polish code.